generator/mavgen_c.py

```diff
@@ -384,6 +384,10 @@ def generate_message_h(directory, m):
  */
 static inline ${return_type} mavlink_msg_${name_lower}_get_${name}(const mavlink_message_t* msg${get_arg})
 {
+    if (msg->len <= ${wire_offset}) {
```

Comment:
I wonder if the spec says that if you trim some zeroes then you must trim them all.  My guess is that it doesn't.

I thought this was previously checking if the field offset was past the end of the message?

Reply:
I'm not sure if I understood your point/question here..
Yes, I think nothing prevents the sender to trim only some of the zero bytes in the end, instead of trimming all of them.

The trimming process reduces the actual message length accordingly, so the receiver then just append the remaining missing zeros into the end of the received message in case the length of the received message is less than what is defined/specified for the particular message type.
Do you think there is still something missing/wrong?

Reply:
Apologies for the slow response.

My concern is that the check you have here doesn't account for the length of the field.  The wire offset may be within the length but the wire-offset-plus-field-length may be outside.  I was thinking in terms of arrays, but I guess anything longer than a byte might be a problem.

Reply:
You are absolutely right. It does not work well in case of longer field types that have zeros in latter bytes (MSB or LSB depending on endianess)

Reply:
Right, this is a bit tricky, implementing partial copy for all possible types and arrays would not sound very handy..

I can think of some more options:

1. Leave the mavlink_msg_<name>_get_<name> function untouched and do an intermediate copy from original msg to full length zero initialized tmp_msg and perform the above function call from that tmp_msg. Then the zeros are already in place and we are not reading from the end of the allocated space.
What to do if malloc is not successful (out of memory case)?

2. Define compiler flag to have possibility to disable sender trim-feature in case system has receiver with MAVLINK_NEED_BYTE_SWAP || !MAVLINK_ALIGNED_FIELDS defined. This is not bullet-proof, because the sender may not be configurable by developer of the receiver.

3. Get rid of the trim feature altogether. How much benefit it gives? At least there are not so many messages with "trimmable" extended fields and not so many bytes to be trimmed from them either. The feature decreases the amount of data transferred over some media, but causes extra load due to memcopy when byte swap not needed and possible incorrect data when byte swap needed.

4) Some new idea to fix this?

Personally I would vote for the option 3).